### PR TITLE
Fix CSharp Payment Example

### PR DESCRIPTION
**Pull request by StephenJosey**
Let me know if anything should be added/changed, thanks!

### CODE REVIEW

File: connect-examples/v2/csharp_payment/Pages/ProcessPayment.cshtml.cs [hunk 1]
```diff
@@ -1,39 +1,31 @@
...
 
-        protected void Page_Load(object sender, EventArgs e)
-        {
+        public ProcessPaymentModel(IConfiguration configuration) {
+            this.LocationId = configuration["AppSettings:LocationId"];
         }
 
-        [System.Web.Services.WebMethod]
-        public static string Charge(string nonce)
+        public void OnPost()
```
Comment (shaofu88) [inline connect-examples/v2/csharp_payment/Pages/ProcessPayment.cshtml.cs]: I think it is a good practice to return a status code here based on response.

File: connect-examples/v2/csharp_payment/appsettings.Development.json [hunk 2]
```diff
@@ -0,0 +1,10 @@
+﻿{
```
Comment (shaofu88) [inline connect-examples/v2/csharp_payment/appsettings.Development.json]: Is this file needed? If so do you want to include AppSettings?

File: connect-examples/v2/csharp_payment/README.md [hunk 3]
```diff
@@ -1,57 +1,57 @@
...
 
-       Update-Package -reinstall -ProjectName CSharpPaymentExample
+    dotnet build
 
 ### Provide required credentials
 
-Both `CSharpPaymentExample/Default.aspx` and `CSharpPaymentExample/Default.aspx.cs` have values near the top of the file
-that you need to replace with various credentials associated with your application.
+Open './appsettings.json' and replace "AccessToken", "LocationId" and "ApplicationId" with the ids you get from your square application created in [Square Developer Portal](https://connect.squareup.com/apps).
+<b>WARNING</b>: never upload .env with your credentials/access_token.
```
[the remark this answers is not in the document]
Comment (StephenJosey) [reply connect-examples/v2/csharp_payment/README.md]: Good catch, I had copied the message from the Rails app. Will be fixed.

File: connect-examples/v2/csharp_payment/Pages/ProcessPayment.cshtml.cs (hunk 1 above)
Comment (hukid) [reply connect-examples/v2/csharp_payment/Pages/ProcessPayment.cshtml.cs]: I don't think this is necessary in razor page handler, the response code is handled internally based on the pagemodel methods invoked

File: connect-examples/v2/csharp_payment/Pages/ProcessPayment.cshtml.cs [hunk 4]
```diff
@@ -54,15 +46,15 @@ public static string Charge(string nonce)
...
             }
             catch (ApiException e)
             {
-                return e.Message;
+                this.ResultMessage = e.Message;
             }
         }
 
-        public static string NewIdempotencyKey()
+        private static string NewIdempotencyKey()
```
Comment (hukid) [inline connect-examples/v2/csharp_payment/Pages/ProcessPayment.cshtml.cs]: Where is this method called?

File: connect-examples/v2/csharp_payment/Pages/ProcessPayment.cshtml.cs [hunk 5]
```diff
@@ -44,7 +36,7 @@ public static string Charge(string nonce)
             // Monetary amounts are specified in the smallest unit of the applicable currency.
             // This amount is in cents. It's also hard-coded for $1.00,
             // which isn't very useful.
-            Money amount = new Money(100, Money.CurrencyEnum.USD);
+            Money amount = new Money(Convert.ToInt32(amountString), Money.CurrencyEnum.USD);
```
Comment (hukid) [inline connect-examples/v2/csharp_payment/Pages/ProcessPayment.cshtml.cs]: Can you directly use integter here?

File: connect-examples/v2/csharp_payment/README.md [hunk 6]
```diff
@@ -1,57 +1,57 @@
 # Payment processing example: Csharp
 
 This sample demonstrates processing card payments with Square Connect API, using the
-Square Connect C# client library.
-
-## Frameworks supported
-- .NET 4.5 or later
+Square Connect C# client library and dotnet core.
```
Comment (hukid) [inline connect-examples/v2/csharp_payment/README.md]: I don't think mention dotnet core is necessary here. but not a strong opinion.

Comment (hukid) [reply connect-examples/v2/csharp_payment/README.md]: instead, you can keep the "## Frameworks supported" with dotnet core 2.0

File: connect-examples/v2/csharp_payment/Startup.cs [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+            else
+            {
+                app.UseExceptionHandler("/Error");
+            }
+
+            app.UseStaticFiles();
+
+            app.UseMvc();
+
+            Square.Connect.Client.Configuration.Default.AccessToken = this.Configuration["AppSettings:AccessToken"];
```
Comment (hukid) [inline connect-examples/v2/csharp_payment/Startup.cs]: Should we add a comment here to explain why we are doing this?

**Comment by hukid** [comment]
Can you append a screenshot for the payment page?

File: connect-examples/v2/csharp_payment/appsettings.Development.json (hunk 2 above)
Comment (StephenJosey) [reply connect-examples/v2/csharp_payment/appsettings.Development.json]: Removed

**Comment by StephenJosey** [comment]
@hukid Screenshot of form (same as the Rails example so please let me know if it should be changed): 
screen shot 2018-07-20 at 11 34 47 am

File: connect-examples/v2/csharp_payment/Pages/ProcessPayment.cshtml.cs (hunk 4 above)
Comment (StephenJosey) [reply connect-examples/v2/csharp_payment/Pages/ProcessPayment.cshtml.cs]: In `OnPost()`: `string uuid = NewIdempotencyKey();`

File: connect-examples/v2/csharp_payment/Pages/ProcessPayment.cshtml.cs (hunk 5 above)
Comment (StephenJosey) [reply connect-examples/v2/csharp_payment/Pages/ProcessPayment.cshtml.cs]: Updated!

File: connect-examples/v2/csharp_payment/Startup.cs (hunk 7 above)
Comment (StephenJosey) [reply connect-examples/v2/csharp_payment/Startup.cs]: Sure! Will this suffice:
// This is globally setting the Square's API access token. 
// This access token will be used for all calls within the application.
